controllers/slicegateway/slicegateway.go

```diff
@@ -918,24 +918,63 @@ func (r *SliceGwReconciler) reconcileGatewayEndpoint(ctx context.Context, sliceG
 		}
 		return true, ctrl.Result{}, err
 	}
-	// endpoint already exists , check if sliceGatewayRemoteNodeIp is changed then update the endpoint
+	// endpoint already exists, check if sliceGatewayRemoteNodeIp is changed then update the endpoint
 	toUpdate := false
 	debugLog.Info("SliceGatewayRemoteNodeIP", "SliceGatewayRemoteNodeIP", sliceGw.Status.Config.SliceGatewayRemoteNodeIPs)
-	if !validateEndpointAddresses(endpointFound.Subsets[0], sliceGw.Status.Config.SliceGatewayRemoteNodeIPs) {
-		debugLog.Info("Updating the Endpoint, since sliceGatewayRemoteNodeIp has changed", "from endpointFound", endpointFound.Subsets[0].Addresses[0].IP)
+
+	restartGWPods := false
+	if nodeIPsCompletelyDifferent(endpointFound.Subsets[0], sliceGw.Status.Config.SliceGatewayRemoteNodeIPs) {
+		// Restart the gateway pods only when the new node IPs are completely distinct
+		restartGWPods = true
+	}
+	if !validateNodeIPsInEndpoint(endpointFound.Subsets[0], sliceGw.Status.Config.SliceGatewayRemoteNodeIPs) {
+		// endpoints gettings used should match with SliceGatewayRemoteNodeIPs
+		log.Info("Updating the Endpoint, since sliceGatewayRemoteNodeIp has changed", "from endpointFound", endpointFound.Subsets[0].Addresses[0].IP)
 		endpointFound.Subsets[0].Addresses = getAddrSlice(sliceGw.Status.Config.SliceGatewayRemoteNodeIPs)
 		toUpdate = true
 	}
+	// When "toUpdate" is set to true we update the endpoints addresses
+	// When "restartGWPods" is set to true, we refresh the connections by restarting the gateway pods
 	if toUpdate {
 		err := r.Update(ctx, &endpointFound)
 		if err != nil {
 			log.Error(err, "Error updating Endpoint")
 			return true, ctrl.Result{}, err
 		}
+		if restartGWPods {
```

Comment:
if nodeIPsCompletelyDifferent(){
   ...
}
we could prevent the usage of extra variable!